src/components/text-editor/prosemirror-adapter/plugins/trigger-plugin.ts

```diff
@@ -0,0 +1,303 @@
+// import { ListItem } from 'src/components/list/list-item.types';
+
+import { Node, NodeType, Schema } from 'prosemirror-model';
+import { Plugin, PluginKey, EditorState } from 'prosemirror-state';
+import { ReplaceStep, ReplaceAroundStep } from 'prosemirror-transform';
+import { EditorView } from 'prosemirror-view';
+import { CustomElement } from 'src/interface';
+
+export const triggerPluginKey = new PluginKey('triggerPlugin');
+
+export type TriggerCharacter =
+    | '@'
+    | '#'
+    | '$'
+    | '!'
+    | '?'
+    | '&'
+    | '*'
+    | '%'
+    | '+'
+    | '-'
+    | '='
+    | '/'
+    | '\\'
+    | '^'
+    | '~'
+    | '`'
+    | ':'
+    | ';'
+    | '|'
+    | '.'
+    | ','
+    | '<'
+    | '>'
+    | '['
+    | ']'
+    | '{'
+    | '}'
+    | '('
+    | ')'
+    | "'";
+
+export type CustomElementInstance = Omit<CustomElement, 'attributes'> & {
+    attributes: Record<string, any>;
+};
```

Comment:
Why do we need to re-type this? :thinking:

Reply:
In defining a `CustomElement` you will have the attributes as a list of strings. When you are actually "instantiating" it you want the attributes as a record. :)

Reply:
I see. I don't like the name `CustomElementInstance` :smile: Why does it need the `Instance` suffix?

Reply:
I will change the name of `CustomElement` to `CustomElementDefinition` and simply call the new on `CustomElement`